comments/board.py

```diff
@@ -0,0 +1,441 @@
+            print(check[1])
+
+    def get_state(self):
+        return self.xy_coord
+
+    def clear_board(self):
+        for all_keys in self.xy_coord:
+            self.xy_coord[all_keys] = None
+
+
+class Checkers_Board(Board):
+    def __init__(self):
+        super().__init__((8, 8))
+        # List of white/black objects
+        self.white_pieces = self._create_pieces("white")  # List
+        self.black_pieces = self._create_pieces("black")  # List
+        self.all_king_rows = self._create_king_row()
+        self.white_side_king_row = self.all_king_rows[0]  # This is (x, 8) row list
+        self.black_side_king_row = self.all_king_rows[1]  # This is (x, 1) row list
+
+    def board_setup(self):  # Place all Checker Pieces
+        white_start_coord_unsort = []
+        black_start_coord_unsort = []
+        for location in self.xy_coord:
+            if location[1] >= 6:  # Top of Board. White Pieces
+                if location[0] % 2 == 1 and location[1] % 2 == 1:
+                    white_start_coord_unsort.append(location)
+                if location[0] % 2 == 0 and location[1] % 2 == 0:
+                    white_start_coord_unsort.append(location)
+            if location[1] <= 3:  # Bottom of Board. Black Pieces
+                if location[0] % 2 == 1 and location[1] % 2 == 1:
+                    black_start_coord_unsort.append(location)
+                if location[0] % 2 == 0 and location[1] % 2 == 0:
+                    black_start_coord_unsort.append(location)
+        white_start_coord = sorted(white_start_coord_unsort, key=lambda y: y[1])
+        black_start_coord = sorted(black_start_coord_unsort, key=lambda y: y[1], reverse=True)
+
+        for i in range(0, len(white_start_coord)):  # White Piece Placement
+            white_xy_coord = (white_start_coord[i])
+            white_piece = self.white_pieces[i]
+            self._place_at_location(white_xy_coord, white_piece)
+            self.white_pieces[i].update_position(white_xy_coord)
+
+        for i in range(0, len(black_start_coord)):  # Black Piece Placement
+            black_xy_coord = (black_start_coord[i])
+            black_piece = self.black_pieces[i]
+            self._place_at_location(black_xy_coord, black_piece)
+            self.black_pieces[i].update_position(black_xy_coord)
+        self.visual = self._generate_visual_board()  # Update visual board in terminal
+
+    def _create_pieces(self, name):
+        piece_list = []
+        for i in range(1, 13):
+            obj = Checkers_Game_Piece(name, i)
+            piece_list.append(obj)
+        return piece_list
+
+    def _create_king_row(self):
+        white_side_king_row = []
+        black__side_king_row = []
+        for x_coord in range(1,9):
+            white_side_king_row.append((x_coord, 8))
+            black__side_king_row.append((x_coord, 1))
+
+        return white_side_king_row, black__side_king_row
+
+    def is_regular_move_valid(self, piece):  # Valid Non-King moves
```

Comment:
I feel like this whole function isn't really necessary.

You use it twice in main.py. Once in process_piece_selection() and once in check_capturable_moves(). But in both instances I feel like you could just update the piece's moves and check if the move is valid or not.

The one thing that this function gives you is that it tells you what are valid directions each piece can go... But why not add that logic directly into the CheckersGamePiece class? Then once it becomes a king just add the two missing moves it needs.